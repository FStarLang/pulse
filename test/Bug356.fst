module Bug356
#lang-pulse
open Pulse.Nolib

fn test (b : bool)
  returns x : UInt32.t
{
  if (b) {
    1ul;
  } else {
    admit();
    2ul;
  }
}

fn test_final_admit (b : bool)
  returns x : UInt32.t
{
  if (b) {
    1ul;
  } else {
    admit();
  }
}