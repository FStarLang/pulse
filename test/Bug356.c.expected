/* krml header omitted for test repeatability */


#include "Bug356.h"

uint32_t Bug356_test(bool b)
{
  if (b)
    return 1U;
  else
  {
    KRML_HOST_EPRINTF("KaRaMeL abort at %s:%d\n%s\n", __FILE__, __LINE__, "");
    KRML_HOST_EXIT(255U);
  }
}

uint32_t Bug356_test_final_admit(bool b)
{
  if (b)
    return 1U;
  else
  {
    KRML_HOST_EPRINTF("KaRaMeL abort at %s:%d\n%s\n", __FILE__, __LINE__, "");
    KRML_HOST_EXIT(255U);
  }
}

